elide-core/src/main/java/com/yahoo/elide/audit/LogMessage.java

```diff
@@ -70,30 +72,57 @@ public LogMessage(String template,
             PersistentResource record,
             String[] expressions,
             int code) throws InvalidSyntaxException {
+        this.template = template;
+        this.record = record;
+        this.expressions = expressions;
         this.operationCode = code;
-        SimpleContext ctx = new SimpleContext();
+    }
 
-        /* Create a new lineage which includes the passed in record */
-        ResourceLineage lineage = new ResourceLineage(record.getLineage(), record);
+    /**
+     * Gets operation code.
+     *
+     * @return the operation code
+     */
+    public int getOperationCode() {
+        return operationCode;
+    }
 
-        for (String name : lineage.getKeys()) {
-            List<PersistentResource> values = lineage.getRecord(name);
+    /**
+     * Gets message.
+     *
+     * @return the message
+     */
+    public String getMessage() {
+        SimpleContext ctx = new SimpleContext();
 
-            ValueExpression expression;
-            if (values.size() == 1) {
-                expression = expressionFactory.createValueExpression(values.get(0).getObject(), Object.class);
-            } else {
-                List<Object> objects = values.stream().map(PersistentResource::getObject).collect(Collectors.toList());
-                expression = expressionFactory.createValueExpression(objects, List.class);
+        if (record != null) {
+            /* Create a new lineage which includes the passed in record */
+            ResourceLineage lineage = new ResourceLineage(record.getLineage(), record);
+
+            for (String name : lineage.getKeys()) {
+                List<PersistentResource> values = lineage.getRecord(name);
+
+                ValueExpression expression;
+                if (values.size() == 1) {
+                    expression = expressionFactory.createValueExpression(values.get(0).getObject(), Object.class);
+                } else {
+                    List<Object> objects = values.stream().map(PersistentResource::getObject)
+                            .collect(Collectors.toList());
+                    expression = expressionFactory.createValueExpression(objects, List.class);
+                }
+                ctx.setVariable(name, expression);
             }
-            ctx.setVariable(name, expression);
+
+            // Add the OpaqueUser to available expressions as "user"
```

Comment:
I'm not sure we want to do this.  This might eclipse a User bean that is in the lineage.  User should be passed to the Logger a different way.